hw18/Lesson18.py

```diff
@@ -0,0 +1,142 @@
+# create class Bot
+class Bot:
+    def __init__(self, name):
+        self.name = name
+
+    def say_name(self):
+        print(self.name)
+
+    def send_message(self, message):
+        print(message)
+
+
+some_bot = Bot('Marvin')
+some_bot.say_name()
+some_bot.send_message('Hello guys')
+
+# Create class TelegramBot inherited from Bot
+
+
+class TelegramBot(Bot):
+
+    def __init__(self, name, url=None, chat_id=None):
+        self.url = url
+        self.chat_id = chat_id
+        name = super().__init__(name) # Which of those 2 rows should I use?
+        # Bot.__init__(self, name)    # Which of those 2 rows should I use?
+
+    def send_message(self, message):
+        print(f'{self.name} bot says {message} to chat {self.chat_id} using {self.url}')
+
+    def set_url(self, url):
+        self.url = url
+
+    def set_chat_id(self, chat_id):
+        self.chat_id = chat_id
+
+
+telegram_bot = TelegramBot('TG')
+
+telegram_bot.say_name()
+telegram_bot.send_message('Hello')
+telegram_bot.set_chat_id(1)
+telegram_bot.send_message('Hello')
+telegram_bot.set_chat_id(2)
+telegram_bot.send_message('Bye')
+
+# Create a class MyStr(str), which overrides the __str__ method
+
+
+class MyStr(str):
+
+    def __str__(self):
+        return self.upper()
+
+
+my_str = MyStr('test')
+print(my_str)
+
+# Create a class User, which overrides the __eq__ method
+
+
+class User:
+    def __init__(self, name):
+        self.name = name
+
+    def __eq__(self, other):
+        return self.name == other.name.upper()
```

Comment:
it won't work if the first name is lowercase or starts with capital letter
```suggestion
        return self.name.upper() == other.name.upper()
```